backend/src/model/entity/plants_impl.rs

```diff
@@ -14,16 +16,43 @@ impl Plants {
     ///
     /// # Errors
     /// * Unknown, diesel doesn't say why it might error.
-    pub fn find_all(conn: &mut PgConnection) -> QueryResult<Vec<PlantsDto>> {
+    pub fn find_all(conn: &mut PgConnection) -> QueryResult<Vec<PlantsSearchDto>> {
         let query_result = plants::table.select(all_columns).load::<Self>(conn);
         query_result.map(|v| v.into_iter().map(Into::into).collect())
     }
 
+    /// Search all plants whose name or species name contains the user provided query string.
+    /// To save traffic, the maximum number of results is limited.
+    ///
+    /// # Errors
+    /// * Unknown, diesel doesn't say why it might error.
+    pub fn search(
+        query: &QueryParameters,
+        conn: &mut PgConnection,
+    ) -> QueryResult<Vec<PlantsSearchDto>> {
+        let capitalized_query = query.search_term.to_lowercase();
+        let query_with_placeholders = format!("%{capitalized_query}%");
+
+        let query = plants::table
+            .select(all_columns)
+            .filter(
+                sql::<Bool>("LOWER(binomial_name) LIKE ")
+                    .bind::<Text, _>(&query_with_placeholders)
+                    .sql(" OR LOWER(ARRAY_TO_STRING(common_name, ' ')) LIKE ")
+                    .bind::<Text, _>(&query_with_placeholders),
+            )
```

Comment:
If we have a powerful ORM we should use its features.
Only if we can't express our statement or they would be inefficient should we fall back to raw SQL.

There is no ned for `to_lowercase`, use ilike instead of like.
For filtering on the binomial name you can do this `.filter(binomail_name.ilike(query_with_placeholders))`.
https://docs.diesel.rs/master/diesel/expression_methods/trait.PgTextExpressionMethods.html#method.ilike

I'm not sure how to achieve something similar for the values in `common_name` but it can probably be done with the querybuilder as well.

Reply:
I opted to use raw SQL strings because I could not get the "correct" diesel ORM-methods to work.
Tbh. I can't quite remember what the exact problem was, but there was a trait missing somewhere. 

That being sain, I agree with your points and would prefer these changes if someone could get them to work.

Reply:
Did it :)